apps/ui/src/xchain/eth/EthereumStorageReactContext.tsx

```diff
@@ -0,0 +1,53 @@
+import { RequiredAsset } from '@force-bridge/commons';
+import { BridgeTransactionStatus, UnFailedTransactionSummary } from '@force-bridge/commons/lib/types/apiv1';
+import { useLocalStorage } from '@rehooks/local-storage';
+import React, { createContext, useContext } from 'react';
+
+export interface EthereumStorage {
+  transactions: TransactionSummaryWithSender[] | null;
+  addTransaction: (tx: EthereumTransaction) => void;
+}
+
+export type TransactionSummaryWithSender = UnFailedTransactionSummary & { sender: string };
+
+export interface EthereumTransaction {
```

Comment:
Maybe we should use a different structure to distinguish sent txs

```ts
type ApproveTx = { type: 'approve', txId: string; timestamp: number; sender: string; asset: string }
type LockTx = { type: 'lock', txId: string; timestamp: number; sender: string; fromAsset: string; toAsset: string }

export type EthereumTx = ApproveTx | LockTx ;

// so we can filter only lock transactions in the history list
function useEthereumTxSummaries() {
  const { txs } = useLocalStorage('...')

  txs.filter((tx) => tx.type === 'lock')
}
```

Reply:
Approve-tx was stored in another localStorage with another store-key